apps/src/sites/studio/pages/components/OrderControls.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+const styles = {
+  controls: {
+    float: 'right'
+  },
+  controlIcon: {
+    margin: '0 5px',
+    cursor: 'pointer'
+  }
+};
+
+const OrderControls = React.createClass({
+  propTypes: {
+    move: React.PropTypes.func.isRequired,
+    remove: React.PropTypes.func.isRequired,
+    type: React.PropTypes.oneOf(['GROUP', 'STAGE']).isRequired,
+    position: React.PropTypes.number.isRequired,
+    total: React.PropTypes.number.isRequired
+  },
+
+  handleMoveUp() {
+    if (this.props.position !== 1) {
+      this.props.move(this.props.type, this.props.position, 'up');
+    }
+  },
+
+  handleMoveDown() {
+    if (this.props.position !== this.props.total) {
+      this.props.move(this.props.type, this.props.position, 'down');
+    }
+  },
+
+  handleRemove() {
+    this.props.remove(this.props.type, this.props.position);
+  },
+
+  render() {
+    return (
+      <div style={styles.controls}>
+        <i
+          onMouseDown={this.handleMoveUp}
+          style={styles.controlIcon}
+          className="fa fa-caret-up"
+        />
+        <i
+          onMouseDown={this.handleMoveDown}
+          style={styles.controlIcon}
+          className="fa fa-caret-down"
+        />
+        <i
+          onMouseDown={this.handleRemove}
+          style={styles.controlIcon}
+          className="fa fa-trash"
+        />
+      </div>
+    );
+  }
+});
+
+export default connect(state => ({}), dispatch => ({
+  move(type, position, direction) {
+    dispatch({type: `MOVE_${type}`, position, direction});
```

Comment:
In a world where we have action creators for each of our actions, this will presumably look something more like
```
if (type === 'GROUP') {
  dispatch(removeGroup(position));
} else if {
  //etc
```